### PR TITLE
Build an API

### LINKED ISSUE #233: Building an API

### PR DESCRIPTION
**Pull request by memcmahon**

### CODE REVIEW

File: module4/lessons/Week2/BuildAnAPI.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+### GET /books/1
+
+<section class='call-to-action' markdown='1'>
+**With a partner: Implement an API Endpoint that satisfies these specifications**  
+
+<section class='note' markdown='1'>
+* Path/Method: GET /api/books/1
+* Status Code: 200
+* Content: JSON containing the book with the id '1' (or whatever id is sent in through the path)
+</section>
```
Comment (rtillies) [inline module4/lessons/Week2/BuildAnAPI.md:215]: The formatting appears off in this section for some reason. The bullets do not render correctly.

File: module4/lessons/Week2/BuildAnAPI.md [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+### PUT /books/1
+
+<section class='call-to-action' markdown='1'>
+**With a partner: Implement an API Endpoint that satisfies these specifications**  
+
+<section class='note' markdown='1'>
+* Path/Method: PUT /api/books/1
+* Status Code: 204
+* Request Content: JSON with the ID of a book to edit, and the field(s) to be edited.
+</section>
```
Comment (rtillies) [inline module4/lessons/Week2/BuildAnAPI.md:321]: The formatting appears off in this section for some reason. The bullets do not render correctly.

File: module4/lessons/Week2/BuildAnAPI.md [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+
+### DELETE /books/1
+
+<section class='call-to-action' markdown='1'>
+
+**Individually: Implement an API Endpoint that satisfies these specifications**  
+<section class='note' markdown='1'>
+* Path/Method: DELETE /api/books/1
+* Status Code: 204
+</section>
```
Comment (rtillies) [inline module4/lessons/Week2/BuildAnAPI.md:370]: The formatting appears off in this section for some reason. The bullets do not render correctly.

File: module4/labs/Week2/BuildAnAPI.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+Today, we will be working individually to reinforce our own understanding of the components of an API.  You _should_ reach out to other students and instructors with questions - we can all help each other learn!
+
+By the end of this Lab, you will have a RESTful API that will allow us to CRUD resources for a database with the following ERD:
+
+![Adopt Don't Shop ERD](/assets/images/module4/week2/adoptdontshop_ERD.png)
+
+## Setup
+
+* Fork and clone this [Starter Repo]().
```
Comment (rtillies) [inline module4/labs/Week2/BuildAnAPI.md]: Any progress on the starter repo yet, or will this be built from a previous lesson's lab?

**Review by rtillies — APPROVED** [review]
This is fantastic, especially for someone who barely knows what API stands for (don't quiz me). I left identical comments on a few sections that do not render correctly, but that's about it. Great work!

File: module4/lessons/Week2/BuildAnAPI.md (hunk 1 above)
Comment (memcmahon) [reply module4/lessons/Week2/BuildAnAPI.md:215]: This is how it looks for me - are you seeing something different?

Screenshot 2023-07-17 103143

File: module4/labs/Week2/BuildAnAPI.md (hunk 4 above)
Comment (memcmahon) [reply module4/labs/Week2/BuildAnAPI.md]: whoops!  It is now linked!

**Pull request merged**